src/main/java/com/example/auctionapp/authentication/AuthenticationController.java

```diff
@@ -0,0 +1,56 @@
+package com.example.auctionapp.authentication;
+
+import com.example.auctionapp.security.config.JWTTokenHelper;
+import com.example.auctionapp.user.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.web.bind.annotation.*;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.Principal;
+import java.security.spec.InvalidKeySpecException;
+
+@RestController
+@RequestMapping("/")
+@CrossOrigin(origins = "http://localhost:3000")
+public class AuthenticationController {
+
+    @Autowired
+    private AuthenticationManager authenticationManager;
+
+    @Autowired
+    private UserDetailsService userDetailsService;
+
+    @Autowired
+    private JWTTokenHelper jwtTokenHelper;
+
+    @PostMapping("api/v1/login")
+    public ResponseEntity<?> login(@RequestBody AuthenticationRequest authenticationRequest) throws InvalidKeySpecException, NoSuchAlgorithmException {
+        final Authentication authentication = authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(authenticationRequest.getEmail(), authenticationRequest.getPassword()));
+
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        User user = (User) authentication.getPrincipal();
+        String jwtToken = jwtTokenHelper.generateToken(user.getEmail());
+
+        AuthenticationResponse response = new AuthenticationResponse();
+        response.setToken(jwtToken);
```

Comment:
Move this to authenticationService. Controller should have as little logic as possible